enterprise/views.py

```diff
@@ -106,12 +122,54 @@ def get_note(provider, required):
                 " not be able to get any discounts from {provider}."
             ).format(provider=provider)
 
-    def get(self, request):
+    @staticmethod
+    def get_course_note(provider, course_name):
         """
-        Render a form to collect user input about data sharing consent.
+        Get a course-specific note for the form.
+
+        Arguments:
+            provider: The name of the linked EnterpriseCustomer
+            course_name: The name of the course in question
+        """
+        return _(
+            "Data sharing consent is required for courses provided by {provider}, including "
+            "{course_name}; if consent is not provided, you will be returned to the Dashboard."
+        ).format(provider=provider, course_name=course_name)
+
+    def get_course_specific_consent(self, request, course_id):
+        """
+        Render a form with course-specific information about data sharing consent.
+        """
+        try:
+            client = CourseApiClient()
+            course_details = client.get_course_details(course_id)
+        except Exception:
+            raise Http404
+        next_url = request.GET.get('next')
+
+        customer = get_object_or_404(EnterpriseCustomerUser, user_id=request.user.id).enterprise_customer
```

Comment:
Not in a database-guaranteed way, but the application logic prevents linking a user to multiple EnterpriseCustomers.

Right now, I'm not sure there's a good way to handle the case where there are multiple EnterpriseCustomerUsers anyway; the form can only handle providing consent to a single one of them.

I think I can narrow down the _potential_ for issues by doing this; let me know what you think:

```python
customer = get_object_or_404(
    EnterpriseCourseEnrollment,
    enterprise_customer_user__user_id=request.user.id,
    course_id=course_id
).enterprise_customer_user.enterprise_customer
```